test/integration_test.py

```diff
@@ -60,16 +61,17 @@ def test_integration_3D(self):
                                  rtol=self.relative_tolerance))
 
     def test_integration_spatial_fit(self):
-        bkg_maker = Grid3DAcceptanceMapCreator(energy_axis=self.energy_axis,
-                                               offset_axis=self.offset_axis,
-                                               oversample_map=5,
-                                               exclude_regions=self.exclude_region_PKS_2155,
-                                               method='fit')
+        bkg_maker = SpatialFitAcceptanceMapCreator(energy_axis=self.energy_axis,
+                                                   offset_axis=self.offset_axis,
+                                                   oversample_map=5,
+                                                   exclude_regions=self.exclude_region_PKS_2155)
         background_model = bkg_maker.create_acceptance_map(observations=self.obs_collection_pks_2155)
         assert type(background_model) is Background3D
+
         reference = Background3D.read('ressource/test_data/reference_model/pks_2155_spatial_fit_bkg.fits')
-        for i in range(background_model.data.shape[0]):
-            print(np.sum((np.abs(background_model.data[i, : ,:] - reference.data[i, : ,:]) / reference.data[i, : ,:]) > 1e-3))
+        relative_error = np.abs(background_model.data - reference.data) / reference.data
+        if np.sum(relative_error > 1e-4) > 0:
+            logging.warning(f'Maximum relative error : {np.nanmax(relative_error)}, fraction above 1e-3 : {np.sum(relative_error > 1e-3)/relative_error.size}, fraction above 1e-2 : {np.sum(relative_error > 1e-2)/relative_error.size}, fraction above 1e-1 : {np.sum(relative_error > 1e-1)/relative_error.size}')
```

Comment:
Maybe we should have an error if the test is too for from expectations insted of a warning.

Reply:
The assert is there for this purpose, the reason for the warning is to have a bit more detailed when the test is failing and not just a true or false. I have actually generalized and improved this a bit in PR #56 .